mailbox/store/src/main/java/org/apache/james/mailbox/store/quota/ListeningCurrentQuotaUpdater.java

```diff
@@ -75,63 +75,66 @@ public boolean isHandling(Event event) {
     }
 
     @Override
-    public void event(Event event) throws MailboxException {
+    public Publisher<Void> reactiveEvent(Event event) {
+        try {
             if (event instanceof Added) {
                 Added addedEvent = (Added) event;
                 QuotaRoot quotaRoot = quotaRootResolver.getQuotaRoot(addedEvent.getMailboxId());
```

Comment:
the quotaroot resolving should be included in the Mono, don't you think?

Reply:
Even better: follow up commits do quota root resolving in a reactive fashion.